spec/lib/vmdb/settings_spec.rb

```diff
@@ -202,4 +214,25 @@
 
     include_examples "password handling"
   end
+
+  describe ".for_resource" do
+    let(:server) { FactoryGirl.create(:miq_server) }
+
+    it "without database changes" do
+      settings = Vmdb::Settings.for_resource(server)
+      expect(settings.api.token_ttl).to eq "10.minutes"
+    end
+
+    it "with database changes" do
+      server.settings_changes.create!(:key => "/api/token_ttl", :value => "2.minutes")
+      settings = Vmdb::Settings.for_resource(server)
+      expect(settings.api.token_ttl).to eq "2.minutes"
+    end
+
+    it "with database changes on an Array" do
+      server.settings_changes.create!(:key => "/log/collection/current/pattern", :value => ["*.log"])
```

Comment:
So, stubbing settings is an incredible pain in the neck, particularly because we build new Settings constants on the fly when things change.  I have a helper method that stubs the Settings const as well as for_resource, but then that, of course, bypasses all the database stuff, which these methods are trying to test directly

Basically, I'm not really sure how to test the Settings machinery by stubbing the Settings machinery :stuck_out_tongue_closed_eyes: